In [4]:
import os
os.chdir("/Users/srujansingh/Documents/JeanFan_Lab/Kidney_Structural_Analysis/")

In [5]:
import scanpy as sc
ctrl1_data = sc.read_h5ad("data/Ctrl_1A2_visiumhd_2um_proximal_labeled.h5ad")

AnnData object is a standardized data structure used in SC and ST datasets that hold all information at one pleace like:

Observations (obs) → cells / spots / pixels

Variables (var) → genes

Matrix (X) → expression values (genes × cells/spots)

Extra metadata → spatial coordinates, annotations, embeddings, etc.


In [ ]:
#Visualizing the AnnData structure:
ctrl1_data

AnnData object with n_obs × n_vars = 5549563 × 19059
    obs: 'in_tissue', 'array_row', 'array_col', 'pxl_row_in_fullres', 'pxl_col_in_fullres', 'proximal_tubule_geojson_feature_index'
    var: 'gene_ids', 'feature_types', 'genome'
    obsm: 'spatial'

The ctr1_data AnnData object contaisn 19059 genes (features) x 5549563 bins (spatial units).

In [ ]:
#Visualizing the observations:
ctrl1_data.obs.head()

,in_tissue,array_row,array_col,pxl_row_in_fullres,pxl_col_in_fullres,proximal_tubule_geojson_feature_index
s_002um_00000_02609-1,1,0,2609,1164.609932,6249.759373,<NA>
s_002um_00000_02610-1,1,0,2610,1164.662587,6245.220638,<NA>
s_002um_00000_02611-1,1,0,2611,1164.715242,6240.681902,<NA>
s_002um_00000_02612-1,1,0,2612,1164.767897,6236.143167,<NA>
s_002um_00000_02613-1,1,0,2613,1164.820552,6231.604433,<NA>


In [25]:
#Visualizing the features:
ctrl1_data.var.head()

,gene_ids,feature_types,genome
Xkr4,ENSMUSG00000051951,Gene Expression,mm10
Rp1,ENSMUSG00000025900,Gene Expression,mm10
Sox17,ENSMUSG00000025902,Gene Expression,mm10
Lypla1,ENSMUSG00000025903,Gene Expression,mm10
Tcea1,ENSMUSG00000033813,Gene Expression,mm10


In [8]:
ctrl1_data.obs["proximal_tubule_geojson_feature_index"].unique()

<IntegerArray>
[<NA>,  929,  145, 2875, 5680, 2409, 2683,   86, 4813,  509,
 ...
 2607, 1603, 4799, 1242, 2392, 3531, 6605, 5953, 1777, 5938]
Length: 1598, dtype: Int64

1.BRUSH BORDER (APICAL MARKERS)

A.Transporters / channels<br>

Slc34a1 – sodium-phosphate cotransporter (classic proximal tubule marker) <br>
Slc5a2 – SGLT2 (glucose reabsorption, very apical)<br>
Slc9a3 – NHE3 (Na⁺/H⁺ exchanger, strongly apical)<br>
Aqp1 – aquaporin-1 (proximal tubule water transport, apical + membrane-wide)<br>

B.Brush border structural / membrane proteins<br>

Vil1 – villin (microvilli actin bundling; strong brush border marker)<br>
Dpp4 – dipeptidyl peptidase 4 (apical membrane enzyme)<br>

2.BASOLATERAL PROXIMAL TUBULE MARKERS:
A.Ion transport / pumps<br>

Atp1a1 – Na⁺/K⁺ ATPase α1 (canonical basolateral pump)<br>
Atp1b1 – Na⁺/K⁺ ATPase β1 subunit<br>
Slc4a4 – NBCe1 (bicarbonate transporter, basolateral acid-base handling)<br>

B.Metabolite / ion exchange<br>

Slc2a1 – GLUT1 (often basolateral glucose transport in certain PT segments)<br>
Slc7a7 – amino acid transport (basolateral exchange roles)


In [22]:
select_genes = ["Slc34a1", "Slc5a2","Slc9a3","Aqp1","Vil1","Dpp4", "Atp1a1","Atp1b1","Slc4a4","Slc2a1","Slc7a7"]
[g in ctrl1_data.var_names for g in select_genes]

[True, True, True, True, True, True, True, True, True, True, True]

In [24]:
#Subsettig the Ctrl1A2 dataset only to a few apical and basal marker genes:
slc=ctrl1_data[:,select_genes]
ctrl_poles=slc.to_df()
ctrl_poles.head()

,Slc34a1,Slc5a2,Slc9a3,Aqp1,Vil1,Dpp4,Atp1a1,Atp1b1,Slc4a4,Slc2a1,Slc7a7
s_002um_00000_02609-1,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
s_002um_00000_02610-1,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
s_002um_00000_02611-1,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
s_002um_00000_02612-1,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
s_002um_00000_02613-1,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
